py4DSTEM/process/diffraction/crystal.py

```diff
@@ -85,22 +85,37 @@ def from_pymatgen_structure(structure, conventional_standard_structure=True):
         """
         import pymatgen as mg
         from pymatgen.ext.matproj import MPRester
-
-        if isinstance(structure, str):
-            with MPRester() as m:
-                structure = m.get_structure_by_material_id(structure)
-
-        assert isinstance(
-            structure, mg.core.Structure
-        ), "structure must be pymatgen Structure object"
-
-        structure = (
-            mg.symmetry.analyzer.SpacegroupAnalyzer(
-                structure
-            ).get_conventional_standard_structure()
-            if conventional_standard_structure
-            else structure
-        )
+        
+        if structure is not None:
+            if isinstance(structure, str):
+                try:
+                    mpr = MPRester(MaPKey)
+                    structure = mpr.get_structure_by_material_id(structure)
+                raise 'Please provide a valid MP API key!!'
```

Comment:
This is a syntax error, so the module doesn't import. I'm not sure why you want to override whatever exception the pymatgen function throws, it should be ok to ditch the `try` block and let that raise the exception. If something other than a bad key were the problem, this would mislead the user.